projects/core/src/auth/client-auth/services/client-token.service.ts

```diff
@@ -0,0 +1,57 @@
+import { Injectable } from '@angular/core';
+import { select, Store } from '@ngrx/store';
+import { Observable } from 'rxjs';
+import { filter, map } from 'rxjs/operators';
+import { LoaderState } from '../../../state/utils/loader/loader-state';
+import { ClientToken } from '../models/client-token.model';
+import { ClientAuthActions } from '../store/actions/index';
+import { StateWithClientAuth } from '../store/client-auth-state';
+import { ClientAuthSelectors } from '../store/selectors/index';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class ClientTokenService {
+  constructor(protected store: Store<StateWithClientAuth>) {}
+
+  /**
+   * Returns a client token. The client token from the store is returned if there is one.
+   * Otherwise a new token is fetched from the backend and saved in the store.
+   */
+  getClientToken(): Observable<ClientToken> {
+    return this.store.pipe(
+      select(ClientAuthSelectors.getClientTokenState),
+      filter((state: LoaderState<ClientToken>) => {
+        if (this.isClientTokenLoaded(state)) {
+          return true;
+        } else {
+          if (!state.loading) {
+            this.store.dispatch(new ClientAuthActions.LoadClientToken());
+          }
+          return false;
+        }
+      }),
+      map((state: LoaderState<ClientToken>) => state.value)
+    );
```

Comment:
`observeOn` added.
`distinctUntilChanged` I think is not needed, because loading states are filtered and also there isn't any long lived subscriptions to it (only some take(1))